nfm-controller/src/reports/publisher_endpoint.rs

```
@@ -59,9 +60,21 @@ where
         credentials_provider: P,
         clock: C,
         compression: ReportCompression,
+        https_proxy: String,
     ) -> Self {
+        let mut builder = Client::builder().use_rustls_tls();
+
+        if !https_proxy.is_empty() {
+            info!("HTTPS proxy configured: {}", https_proxy);
+            if let Ok(proxy) = Proxy::https(&https_proxy) {
+                builder = builder.proxy(proxy);
+            } else {
+                warn!("Invalid HTTPS proxy URL: {}", https_proxy);
+            }
```

When the proxy endpoint is invalid, this prints out a warning every report publication cycle.  However, all configuration is loaded when the application starts.  I feel it would make more sense to validate the proxy URL upfront on startup and fail immediately.

Perhaps similar to how invalid arg combinations are validated on startup, no?  (e.g. `--endpoint` must be specified if `--publish-reports` is on.)

---

Ah, doing so on startup is good, thanks for pointing that out.  I do think the right thing is to fail hard on invalid input.  The application will make sure all the input it has is right, then start doing work.

---

Sounds good. I'll update the pull request to implement this behavior